net/src/parse.rs

```diff
@@ -160,8 +174,23 @@ pub enum ParseError<E: core::error::Error> {
 
 #[derive(thiserror::Error, Debug)]
 pub enum DeParseError<E> {
+    #[error("Deserialization buffer longer than 2^16 bytes ({0} bytes given)")]
+    BufferTooLong(usize),
     #[error(transparent)]
     Length(LengthError),
     #[error(transparent)]
     Invalid(E),
 }
+
+pub trait ToNonZeroUSize {
```

Comment:
Why can't this use the From trait?

Reply:
we absolutely can.  But I thought the theory with this was that the method name was not descriptive enough.

Is `cast` worse than `into`?